pkg/api/types.go

```diff
@@ -266,6 +266,19 @@ type V20170131ARMContainerService struct {
 	*v20170131.ContainerService
 }
 
+// VlabsUpgradeContainerService is the type we read and write from file
+// needed because the json that is sent to ARM and acs-engine
+// is different from the json that the ACS RP Api gets from ARM
+type VlabsUpgradeContainerService struct {
+	TypeMeta
+	*vlabs.UpgradeContainerService
+}
+
+// UpgradeContainerService API model
+type UpgradeContainerService struct {
+	OrchestratorProfile *OrchestratorProfile `json:"orchestratorProfile,omitempty"`
```

Comment:
What about having Properties in there? That's how the regular apimodel works, it has "apiVerson" and "properties" as the two top-level things.

(context: I was trying to test some integration stuff and wanted to cheat and generate with one version and then pass the other versioned apimodel to the upgrade call, but it nil derefs in the switch statement because it doesn't actually deserialize properly since there's no Properties level here)

Not that we are trying to accommodate the apimodel as the upgrade model input, but for consistency sake maybe?